src/Map.js

```diff
@@ -85,38 +91,50 @@ class PresentationMap extends React.PureComponent {
 	}
 
 	render() {
-		const {children, mapComponent, ...props} = this.props;
+		const {children, mapComponent, wrapper, wrapperOptions, ...props} = this.props;
 
 		if (!mapComponent) {
 			return (<Error centered>mapComponent not supplied to Map</Error>);
 		} else {
-
 			if (!props.stateMapKey) {
 				props.view = this.state.view || props.view;
 				props.onViewChange = this.onViewChange;
 			}
 
-			let map = React.createElement(mapComponent, props, children); //todo ptr-map-wrapper ?
+			if (wrapper) {
+				const wrapperComponent = this.props.wrapper.prototype && this.props.wrapper.prototype.isReactComponent ? this.props.wrapper : MapWrapper;
 
-			if (!children) {
-				return map;
-			} else {
-				return (
-					<div className="ptr-map-controls-wrapper">
-						{map}
-						{React.Children.map(children, child => {
-							return React.cloneElement(child, {
-								...child.props,
-								view: this.props.stateMapKey ? this.props.view : (this.state.view || this.props.view),
-								viewLimits: this.props.viewLimits,
-								updateView: this.props.stateMapKey ? this.props.onViewChange : this.onViewChange,
-								resetHeading: this.props.stateMapKey ? this.props.resetHeading : this.resetHeading
-							});
-						})}
-					</div>
+				return React.createElement(
+					wrapperComponent,
+					{...props, ...wrapperOptions},
```

Comment:
If wrapperOptions object is just wrapper props, it should be called wrapperProps